Lib/asyncio/windows_events.py

```diff
@@ -515,7 +517,9 @@ def finish_recv(trans, key, ov):
             try:
                 return ov.getresult()
             except OSError as exc:
-                if exc.winerror in (_overlapped.ERROR_NETNAME_DELETED,
+                if exc.winerror == _overlapped.ERROR_PORT_UNREACHABLE:
+                    return b'', None
```

Comment:
This code is not covered by tests.

Reply:
Yup, I can add that.

Reply:
@serhiy-storchaka
I've made some adjustments to ensure the error check only affects areas where we get results from `WSARecvFrom` (this is closer to my original implementation, but I'd forgotten some of the details since then and so messed it up a bit during the merge from main).

I've also added some tests to `test_sock_lowlevel` to test the `recvfrom_into` path. As far as I could tell, `recvfrom_into` isn't used by the `DatagramTransport` and in order to get the sockets into the particular states they need to be in the test is fairly contrived. I've only applied it to the `ProactorEventLoopTests`.